src/Generators/SubProfileGenerator.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Drupal\lightning\Generators;
+
+use Drupal\lightning\ComponentDiscovery;
+use DrupalCodeGenerator\Command\BaseGenerator;
+use DrupalCodeGenerator\Utils;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Question\Question;
+
+class SubProfileGenerator extends BaseGenerator
+{
+  /**
+   * The Lightning component discovery helper.
+   *
+   * @var \Drupal\lightning\ComponentDiscovery
+   */
+  protected $componentDiscovery;
+
+  /**
+   * Modules to include in the sub-profile.
+   *
+   * @var string[]
+   */
+  protected $installList = [];
+
+  protected $name = 'lightning-subprofile';
```

Comment:
Nit: Can we add `{@inheritdoc}` comments above all these members that we're overriding from BaseGenerator?